x-pack/plugins/apm/public/components/routing/templates/apm_service_template/index.tsx

```diff
@@ -123,12 +124,41 @@ function TemplateWithContext({
   );
 }
 
+export function isMetricsTabHidden({
+  agentName,
+  serviceRuntimeName,
+}: {
+  agentName?: string;
+  serviceRuntimeName?: string;
+}) {
+  return (
+    !agentName ||
+    isRumAgentName(agentName) ||
+    isJavaAgentName(agentName) ||
+    isIosAgentName(agentName) ||
+    isJRubyAgent(serviceRuntimeName)
+  );
+}
+
+export function isJVMsTabHidden({
+  agentName,
+  serviceRuntimeName,
+}: {
+  agentName?: string;
+  serviceRuntimeName?: string;
+}) {
+  return !(isJavaAgentName(agentName) || isJRubyAgent(serviceRuntimeName));
+}
+
 function useTabs({ selectedTab }: { selectedTab: Tab['key'] }) {
-  const { agentName } = useApmServiceContext();
+  const { agentMetadataDetails } = useApmServiceContext();
```

Comment:
Isn't this metadata for the service?

```suggestion
  const { serviceMetadataDetails } = useApmServiceContext();
```

Either way, something rubs me the wrong way about exposing such an opaque thing as "metadata". What about this:

```suggestion
  const { agentName, serviceRuntimeName } = useApmServiceContext();
```